### PR TITLE
Support getitem by ellipsis index in dynamic mode

### PR DESCRIPTION
**Pull request by zyfncg**
### PR types

New features

### PR changes

APIs

### Describe

Support getitem by ellipsis index in dynamic mode

Example:
```
x = paddle.to_tensor(np.array([[[1, 2], [3, 4]], [[5, 6], [7, 8]]]).astype(np.float32))
y = x[..., 0]
```

### CODE REVIEW

**Review by wxzylforever — COMMENTED** [review]
LGTM

**Review by hbwx24 — COMMENTED** [review]
LGTM

File: paddle/fluid/pybind/imperative.cc [hunk 1]
```diff
@@ -467,20 +479,25 @@ static void ParseIndexingSlice(framework::LoDTensor *tensor, PyObject *_index,
...
+        dim++;
         continue;
       }
       slice_axes->push_back(dim);
       slice_starts->push_back(start);
       slice_ends->push_back(end);
       slice_strides->push_back(step);
+      dim++;
+    } else if (slice_item == Py_Ellipsis) {
+      dim += rank - specified_dims;
```
Comment (chenwhql) [inline paddle/fluid/pybind/imperative.cc:493]: may be adding else branch is better, we can move the error hint into else branch, such as
```
} else {
    PADDLE_THROW(
        platform::errors::InvalidArgument(
                  "Currently, VarBase.__getitem__() only allows "
                  "indexing by Integers, Slices, Ellipsis, and tuples of "
                  "these types, but received %s in %dth slice item",
                  std::string(Py_TYPE(slice_item)->tp_name), i + 1));
}
```

File: python/paddle/fluid/tests/unittests/test_var_base.py [hunk 2]
```diff
@@ -652,6 +652,59 @@ def _test_slice_for_tensor_attr(self):
...
+    def _test_for_getitem_ellipsis_index(self):
+        shape = (64, 3, 5, 256)
+        np_fp32_value = np.random.random(shape).astype('float32')
+        np_int_value = np.random.randint(1, 100, shape)
+
+        var_fp32 = paddle.to_tensor(np_fp32_value)
+        var_int = paddle.to_tensor(np_int_value)
+
+        # test for float32
+        var = [
```
Comment (chenwhql) [inline python/paddle/fluid/tests/unittests/test_var_base.py]: duplicate code? put the duplicate code in another func?

File: paddle/fluid/pybind/imperative.cc (hunk 1 above)
Comment (zyfncg) [reply paddle/fluid/pybind/imperative.cc:493]: Done!

File: python/paddle/fluid/tests/unittests/test_var_base.py (hunk 2 above)
Comment (zyfncg) [reply python/paddle/fluid/tests/unittests/test_var_base.py]: Done!

**Review by chenwhql — APPROVED** [review]
LGTM

**Pull request merged**